app/cli/main.go

```diff
@@ -16,25 +16,82 @@
 package main
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"os"
+	"strings"
 
 	"github.com/chainloop-dev/chainloop/app/cli/cmd"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry"
+	"github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/posthog"
 	v1 "github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1"
 	"github.com/go-kratos/kratos/v2/errors"
 	jwtMiddleware "github.com/go-kratos/kratos/v2/middleware/auth/jwt"
 	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc/status"
 )
 
+// Follow the convention stated on https://consoledonottrack.com/
+const doNotTrackEnv = "DO_NOT_TRACK"
+
+var (
+	// Posthog API key and endpoint to be injected in the build process
+	posthogAPIKey   = ""
+	posthogEndpoint = ""
+)
+
 func main() {
 	// Couldn't find an easier way to disable the timestamp
 	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr, FormatTimestamp: func(interface{}) string { return "" }})
 	rootCmd := cmd.NewRootCmd(logger)
 
-	if err := rootCmd.Execute(); err != nil {
-		msg, exitCode := errorInfo(err, logger)
+	// Set the default exit code to 0, in case of an error it will be updated
+	exitCode := 0
+
+	// Run the command and get the executed command
+	executedCmd, err := rootCmd.ExecuteC()
+	if err != nil {
+		var msg string
+		msg, exitCode = errorInfo(err, logger)
 		logger.Error().Msg(msg)
-		os.Exit(exitCode)
+	}
+
+	runTelemetry(logger, executedCmd, exitCode)
+
+	os.Exit(exitCode)
```

Comment:
do we want to `os.Exit` even if we have 0 as returned code? I know it's the last statement of the program but I am not sure if there is an additional consequence of doing this instead of letting the program finish as before

Reply:
I know but the other option I think it's weird:
```golang
if !isTelemetryDisabled() {
  recordCommand(logger, executedCmd, exitCode)
}

if exitCode != 0 {
  os.Exit(exitCode)
}
```

Reply:
yeah, I agree, my question was more out of curiosity, to know if there was a difference between letting the program finish or calling exit(0), it might not be.

Reply:
Ahh sorry! To be honest I'm not sure, I guess it won't have any different :/